src/main/native/GCM.c

```diff
@@ -647,1562 +470,1723 @@ int GCM_encrypt_core(JNIEnv* env, ICC_CTX* ockCtx, ICC_AES_GCM_CTX* gcmCtx,
+                 		NULL, 0,
+                       	data + dataOffset, dataLen,
+                        out + outOffset, updateOutlen);  
+            
+           		
+#ifdef DEBUG_GCM_DETAIL
+            if ( debug ) {                  
+           		gslogMessage ("NI GCM_UpdForUpdateEncrypt_core dataLen > 0 path after ICC_AES_GCM_EncryptUpdate rc %d updateOutlen %ld\n", rc, *updateOutlen);
+            	gslogMessage ("DETAIL_GCM rc ICC_AES_GCM_EncryptUpdate(plaintextLen > 0) %d updateOutlen %d", (int) rc, *updateOutlen);
+            }
+#endif	
+			
+			if (rc != ICC_OSSL_SUCCESS) {
+				ockCheckStatus(ockCtx);
+#ifdef DEBUG_GCM_DETAIL
+        		if ( debug ) {  				
+					gslogMessage ("NI ICC_AES_GCM_CRYPTUPDATE_FAILED\n");
+				}
+#endif				
+				return ICC_AES_GCM_CRYPTUPDATE_FAILED;
+			}
+		}
 	} else {
-		ret = GetPRIMITICEARRAYCRITICAL;
-	}
-
-        if (keyNative != NULL ) {
-                (*env)->ReleasePrimitiveArrayCritical(env, key, keyNative, 0);
-        }
-
-        if (ivNative != NULL ) {
-                (*env)->ReleasePrimitiveArrayCritical(env, iv,  ivNative, 0);
-        }
-
-        if (tagNative != NULL ) {
-                (*env)->ReleasePrimitiveArrayCritical(env, tag, tagNative, 0);
-        }
-
-        if (aadNative != NULL) {
-                (*env)->ReleasePrimitiveArrayCritical(env, aad, aadNative, 0);
-        }
-
-        if (plaintextNative != NULL && plaintextLen > 0) {
-                (*env)->ReleasePrimitiveArrayCritical(env, plaintext, plaintextNative, 0);
-        }
-
-        if (ciphertextNative != NULL ) {
-                (*env)->ReleasePrimitiveArrayCritical(env, ciphertext, ciphertextNative, 0);
-        }
-
-        if( debug ) {
-                gslogFunctionExit(functionName);
-        }
-
-        return (jint)ret;
-}
-//============================================================================
-///*
-// * Class:     com_ibm_crypto_plus_provider_ock_NativeInterface
-//  * Method:    do_GCM_delete
-//   * Signature: (JJ)V
-//    */
-JNIEXPORT void JNICALL Java_com_ibm_crypto_plus_provider_ock_NativeInterface_do_GCM_1delete
-	(JNIEnv *env, jclass thisObj, jlong ockContextId) {
-	static const char * functionName = "NativeInterface.do_GCM_delete";
-	ICC_CTX * ockCtx = (ICC_CTX *)((intptr_t) ockContextId);
-	if	( debug ) {
-		gslogFunctionEntry(functionName);
+		ockCheckStatus(ockCtx);
+#ifdef DEBUG_GCM_DETAIL
+        if ( debug ) {  		
+			gslogMessage ("NI ICC_AES_GCM_CTX_NEW_FAILED\n");
+		}
+#endif		
+		return ICC_AES_GCM_CTX_NEW_FAILED;
 	}
-                     
-	getOrfreeGCMContext(ockCtx, 0);
 
 	if ( debug ) {
 		gslogFunctionExit(functionName);
 	}
+	return 0;
+#else
+      	return -1;
+#endif
 }
+/*============================================================================
+* Class:     com_ibm_crypto_plus_provider_ock_NativeInterface
+* Method:    internal functions
+* Signature: 
+*/
+int GCM_FinalForUpdateDecrypt_core (JNIEnv *env, ICC_CTX* ockCtx, ICC_AES_GCM_CTX* gcmCtx,
```

Comment:
Space before `*`.

Reply:
Should be consistant spacing now surround *.